third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1118,6 +1127,15 @@ struct Frame {
     function: Arc<Function>,
     ty_args: Vec<Type>,
     local_tys: Vec<Type>,
+    ty_cache: FrameCache,
+}
+
+#[derive(Default)]
+struct FrameCache {
+    struct_field_instantiation: HashMap<StructDefInstantiationIndex, Vec<Type>>,
+    struct_def_instantiation: HashMap<StructDefInstantiationIndex, Type>,
+    field_instantiation: HashMap<FieldInstantiationIndex, Type>,
```

Comment:
It seems pretty wasteful to have two separate caches for fields. Can we merge the two?